packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -109,6 +109,31 @@ export const CouponForm = ({
   const [checkingCoupon, setCheckingCoupon] = useState(false);
 
   const onFormMounted = useCallback(() => couponMounted({ planId }), [planId]);
+
+  // check if coupon code was included in URL
+  // if true and valid, apply coupon code on page load
+  useEffect(() => {
+    (async () => {
+      const params = new URLSearchParams(window.location.search);
```

Comment:
Instead of getting the query params from `window` here, could we update the existing params logic found [here](https://github.com/mozilla/fxa/blob/7087e61156e8c4ee61371471329fe52d7e871d49/packages/fxa-payments-server/src/index.tsx#L30)? 

Once the existing queryParms logic is updated, the value should be available in the `AppContext`, so you could just read it from there.